### ISSUE TITLE
Limit use of ARIA in SVGs

### ISSUE THREAD

**Issue by mgifford** [uswds/uswds#4307]
In looking at existing SVGs https://github.com/uswds/uswds/pull/4304

I ran into `aria-hidden="true" focusable="false”` within the SVG. I have no idea why this would be included, but really would like to remove ARIA unless it has a clear specific role.

Defining `role="img"` is useful. Looking at [Carie Fisher's work on Smashing Mag](https://www.smashingmagazine.com/2021/05/accessible-svg-patterns-comparison/) as an example:

```
<svg role="img" aria-labelledby="fox11 description11" ...>
   <title id="fox11">What does the fox say?</title>
   <desc id="description11">Will we ever know?</desc>
   [design code]
</svg>
```

We might want the images to look a bit more like:

```
<svg xmlns="http://www.w3.org/2000/svg" width="64" height="39" viewBox="0 0 64 39" role="img" aria-labelledby="title-arrow1 description-arrow1">
  <title id="title-arrow1">Down angle arrow</title>
  <description id="description-arrow1"></description>
  <style>
    path {
      fill: #005EA2;
   }
   @media (prefers-color-scheme: dark) {
     path {
        fill: #58b4ff;
      }
    }
  </style>
  <g>
    <path d="M63.36 4.004L60.155.741c-.427-.436-.919-.654-1.475-.654-.555 0-1.047.218-1.475.654L32 26.4 6.798.74C6.37.306 5.878.088 5.323.088c-.556 0-1.048.218-1.476.654L.642 4.005C.213 4.44 0 4.941 0 5.507c0 .565.214 1.066.642 1.501l29.883 30.427c.427.435.92.652 1.475.652.556 0 1.047-.217 1.474-.652L63.361 7.008c.427-.435.639-.936.639-1.501 0-.566-.212-1.067-.64-1.503z" />
  </g>
</svg>
```

**Comment by thisisdano** [comment]
I would like to address this, but I know there are some compatibility reasons for our SVG being the way it is and we'll need to research and document the why of our current state before we make any decision here

**Comment by thisisdano** [comment]
So, I believe this code is here because for a couple reasons, initially outlined here: https://github.com/uswds/uswds/pull/3892

The gist is that we expect these icons to be used as decorative images, or as visual progressive enhancements. We don't want them to be selectable or announced as they would be redundant. We would probably prefer to use only `role="img"` and `alt=""`, but we needed to add the `aria-hidden` and `focusable="false"` attributes to fix a couple of browser bugs. 

It's possible that those are no longer bugs. And in the case of `focusable="false"`, the browser is IE11, which we no longer support. But I'm not sure that we'd remove the code to explicitly remove IE11 support if we didn't have to do this.

**Comment by mgifford** [comment]
The best time to remove IE support is a decade ago, the second best time is now. 

Seriously though, if there are going to be images that are more than just images, then the filename should highlight that they are decorative. It should be clear by just looking at the filename, and then it should be clearly documented. Most folks just aren't going to look into the SVG file. 

If this isn't something that can be removed from the current release, it would be good to commit to doing it the next major release (when hopefully the design system isn't limited by IE11 support).

**Comment by werdnanoslen** [comment]
Perhaps related, I see that the lock.svg icon in the [banner](https://designsystem.digital.gov/components/banner/#component-code/) is meant to have an `aria-labelledby` attr. 

As a pattern, should icons be given those aria attributes at all, given the guidance above? If so, should it be given ad hoc / manually as the example shows (rather than simply loading the svg from the file), or should it be in the svg file bundled in the npm package? Currently, it is not in that package (at [`/uswds/src/img/usa-icons/lock.svg`](https://www.npmjs.com/package/uswds?activeTab=code)`).

**Comment by mejiaj** [comment]
Hi @werdnanoslen, 

>As a pattern, should icons be given those aria attributes at all, given the guidance above?

If you're image isn't decorative you should include `role="img"` and label it.


>If so, should it be given ad hoc / manually as the example shows (rather than simply loading the svg from the file), or should it be in the svg file bundled in the npm package? Currently, it is not in that package (at [/uswds/src/img/usa-icons/lock.svg](https://www.npmjs.com/package/uswds?activeTab=code)`).

This icon is from a previous set, which is why it's not currently in `usa-icons`. Lock is available in the template as an inline SVG, but it's also included as a file in `packages/usa-banner/src/img`. I've created this issue #5303 for that.


---

Outstanding tasks for this issue:
1. Confirm if SVGs still require `aria-hidden` and `focusable="false"` (previously for legacy browsers)
2. Confirm #5303.

**Comment by werdnanoslen** [comment]
Right, main question I have is whether to expect (at some point)...
1. The `<svg>`s shipped in the package will contain certain _critical_ attributes other than the paths, like`aria-labelledby`, `<title>`, and `<desc>`
2. Consumers should manually add those attributes to an `img` element whose source is the package svg (containing only the paths)
3. Consumers should import the package `<svg>` element but wrap it in a span/div with those attributes

Pardon if this is going in a different direction than this issue was intended, but seems like a fundamental question.

**Comment by mgifford** [comment]
1.  I agree that SVGs are a mess as far as multilingual content goes. 
2.  Generally, I've tried to default to the most accessible option for most folks. Generally I think duplicated content is better than not showing relevant semantic content. 
3.  A designer could add the `aria-hidden="true"` in the SVG or in a surrounding element. In my experience, most folks aren't comfortable editing the SVG files, even if they know HTML. Just easier to train folks to hide content using HTML explaining to them that an SVG is just a text file that they can use Notepad to edit.   
      
    Hope that helps.

**Comment by annepetersen** [comment]
USWDS is currently understaffed, so I'm closing items that aren’t as high on the priority list right now, including this one, unfortunately.
